lib/acme/client/faraday_middleware.rb

```diff
@@ -14,7 +14,7 @@ def initialize(app, client:)
   def call(env)
     @env = env
     @env[:request_headers]['User-Agent'] = USER_AGENT
-    @env.body = crypto.generate_signed_jws(header: { nonce: pop_nonce }, payload: env.body)
+    @env.body = client.jwk.jws(header: { nonce: pop_nonce }, payload: env.body)
```

Comment:
Its a string but a middleware could misbehave and change it for something else. Its generally assume that its going to be a string.